internal/resource/resource.go

```diff
@@ -268,11 +283,12 @@ func (r *Resource) SnapshotWithOverrides(ctx context.Context, comp *apiv1.Compos
 type Snapshot struct {
 	*Resource
 
-	ReconcileInterval *metav1.Duration
-	Disable           bool
-	DisableUpdates    bool
-	Replace           bool
-	Orphan            bool
+	ReconcileInterval  *metav1.Duration
```

Comment:
Added foreground deletion to work with the new requeue method's PendingForegroundDeletion `pendingForegroundDeletion := (resource != nil && resource.Deleted(comp) && !resource.Disable && resource.ForegroundDeletion)`